### ISSUE TITLE
'reducedDims' is not a slot in class "SingleCellExperiment"

### ISSUE THREAD

**Issue by wandasen** [cole-trapnell-lab/monocle3#272]
I use demo data for testing:
# Load the data
expression_matrix <- readRDS(url("http://staff.washington.edu/hpliner/data/cao_l2_expression.rds"))
cell_metadata <- readRDS(url("http://staff.washington.edu/hpliner/data/cao_l2_colData.rds"))
gene_annotation <- readRDS(url("http://staff.washington.edu/hpliner/data/cao_l2_rowData.rds"))

# Make the CDS object
cds <- new_cell_data_set(expression_matrix,
cell_metadata = cell_metadata,
gene_metadata = gene_annotation)
Then ,An error has occurred:
Error in checkSlotAssignment(object, name, value) :
  'reducedDims' is not a slot in class "SingleCellExperiment"

**Comment by hpliner** [comment]
We believe this is solved in the latest develop version of the software. Please upgrade using `devtools::install_github('cole-trapnell-lab/monocle3', ref="develop")` and let us know if the issue remains!

**Comment by ragak** [comment]
Hi, I had the same issue when I updated monocle3 (I'm on macOS Mojave using RStudio with R3.6.2). The only way I could get it to work again was to downgrade SingleCellExperiment to version 1.6 and downgrade Monocle3 to version 0.2.0. Any idea if this is something I can easily fix? Thank you.

**Comment by dsb66** [comment]
I have the same issue: after updating from BioConductor 3.9 to 3.10, I started receiving the error: 

Error in checkSlotAssignment(object, name, value) : 
  ‘reducedDims’ is not a slot in class “SingleCellExperiment”

with the new_cell_data_set command. The instruction used to work with BioConductor 3.9.

**Comment by dsb66** [comment]
quick update: doing a forced reinstall of monocle3 after upgrading to BioConductor3.10 fixes the error 
Error in checkSlotAssignment(object, name, value) :
‘reducedDims’ is not a slot in class “SingleCellExperiment”

**Comment by akshatkarambe** [comment]
> We believe this is solved in the latest develop version of the software. Please upgrade using `devtools::install_github('cole-trapnell-lab/monocle3', ref="develop")` and let us know if the issue remains!

hello,

I reinstalled it using your code but still, the issue remains.
Is there anything else i could do other than downgrading Monocle3 to 0.2.0?

Best, 
Akshat

**Comment by nfancy** [comment]
Hi I am having the same error.

```
> packageVersion("monocle3")
[1] ‘0.2.1.2’

> sessionInfo()
R version 3.6.3 (2020-02-29)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1

locale:
 [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C               LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
 [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8    LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] KernSmooth_2.23-16 fields_10.3        maps_3.3.0         spam_2.5-1         dotCall64_1.0-0   
 [6] Seurat_3.1.4       dplyr_0.8.5        DT_0.12            plotly_4.9.2       ggplot2_3.3.0     
[11] scFlow_0.4.5      

loaded via a namespace (and not attached):
  [1] rsvd_1.0.3                  ica_1.0-2                   svglite_1.2.3              
  [4] ps_1.3.2                    foreach_1.4.8               lmtest_0.9-37              
  [7] rprojroot_1.3-2             crayon_1.3.4                jcolors_0.0.4              
 [10] MASS_7.3-51.5               MAST_1.12.0                 nlme_3.1-144               
 [13] backports_1.1.5             monocle3_0.2.1.2            rlang_0.4.5                
 [16] XVector_0.26.0              bib2df_1.1.1                ROCR_1.0-7                 
 [19] irlba_2.3.3                 callr_3.4.2                 limma_3.42.2               
 [22] scater_1.14.6               apcluster_1.4.8             BiocParallel_1.20.1        
 [25] rjson_0.2.20                bit64_0.9-7                 glue_1.3.2                 
 [28] rngtools_1.5                sctransform_0.2.1           parallel_3.6.3             
 [31] processx_3.4.2              vipor_0.4.5                 AnnotationDbi_1.48.0       
 [34] BiocGenerics_0.32.0         HGNChelper_0.8.1            tidyselect_1.0.0           
 [37] SummarizedExperiment_1.16.1 usethis_1.5.0               liger_0.5.0.9000           
 [40] fitdistrplus_1.0-14         XML_3.99-0.3                tidyr_1.0.2                
 [43] zoo_1.8-7                   ggpubr_0.2.5                orthopolynom_1.0-5         
 [46] formattable_0.2.0.1         magrittr_1.5                evaluate_0.14              
 [49] bibtex_0.4.2.2              Rdpack_0.11-1               gdtools_0.2.1              
 [52] cli_2.0.2                   zlibbioc_1.32.0             sn_1.5-5                   
 [55] rstudioapi_0.11             doRNG_1.8.2                 whisker_0.4                
 [58] BiocSingular_1.2.2          xfun_0.12                   askpass_1.1                
 [61] pkgbuild_1.0.6              multtest_2.42.0             cluster_2.1.0              
 [64] caTools_1.18.0              doSNOW_1.0.18               KEGGREST_1.26.1            
 [67] tibble_2.1.3                ggrepel_0.8.2               threejs_0.3.3              
 [70] ape_5.3                     listenv_0.8.0               Biostrings_2.54.0          
 [73] png_0.1-7                   future_1.16.0               withr_2.1.2                
 [76] bitops_1.0-6                plyr_1.8.6                  dqrng_0.2.1                
 [79] pillar_1.4.3                RcppParallel_5.0.0          gplots_3.0.3               
 [82] multcomp_1.4-12             fs_1.3.1                    graphite_1.32.0            
 [85] paletteer_1.1.0             DelayedMatrixStats_1.8.0    vctrs_0.2.4                
 [88] ellipsis_0.3.0              devtools_2.0.2              metap_1.3                  
 [91] tools_3.6.3                 beeswarm_0.2.3              munsell_0.5.0              
 [94] DelayedArray_0.12.2         compiler_3.6.3              pkgload_1.0.2              
 [97] abind_1.4-5                 sessioninfo_1.1.1           scico_1.1.0                
[100] GenomeInfoDbData_1.2.2      gridExtra_2.3               enrichR_2.1                
[103] edgeR_3.28.1                riverplot_0.6               lattice_0.20-40            
[106] mutoss_0.1-12               snow_0.4-3                  BiocFileCache_1.10.2       
[109] humaniformat_0.6.0          jsonlite_1.6.1              scales_1.1.0               
[112] graph_1.64.0                pbapply_1.4-2               lazyeval_0.2.2             
[115] doParallel_1.0.15           R.utils_2.9.2               reticulate_1.14            
[118] checkmate_2.0.0             rmarkdown_2.1               sandwich_2.5-1             
[121] cowplot_1.0.0               Rtsne_0.15                  dichromat_2.0-0            
[124] Biobase_2.46.0              uwot_0.1.8                  igraph_1.2.4.2             
[127] HDF5Array_1.14.2            survival_3.1-8              numDeriv_2016.8-1.1        
[130] yaml_2.2.1                  plotrix_3.7-7               systemfonts_0.1.1          
[133] htmltools_0.4.0             memoise_1.1.0               locfit_1.5-9.1             
[136] IRanges_2.20.2              viridisLite_0.3.0           digest_0.6.25              
[139] assertthat_0.2.1            rappdirs_0.3.1              npsurv_0.4-0               
[142] palr_0.2.0                  pals_1.6                    RSQLite_2.2.0              
[145] future.apply_1.4.0          lsei_1.2-0                  mapproj_1.2.7              
[148] remotes_2.1.0               data.table_1.12.8           blob_1.2.1                 
[151] S4Vectors_0.24.3            R.oo_1.23.0                 splines_3.6.3              
[154] labeling_0.3                rematch2_2.1.0              Rhdf5lib_1.8.0             
[157] RCurl_1.98-1.1              hms_0.5.3                   rhdf5_2.30.1               
[160] colorspace_1.4-1            DropletUtils_1.6.1          base64enc_0.1-3            
[163] WebGestaltR_0.4.3           mnormt_1.5-6                ggbeeswarm_0.6.0           
[166] GenomicRanges_1.38.0        Rcpp_1.0.4                  mclust_5.4.5               
[169] RANN_2.6.1                  mvtnorm_1.1-0               fansi_0.4.1                
[172] R6_2.4.1                    ggridges_0.5.2              lifecycle_0.2.0            
[175] DoubletFinder_2.0.2         oompaBase_3.2.9             TFisher_0.2.0              
[178] curl_4.3                    ggsignif_0.6.0              gdata_2.18.0               
[181] leiden_0.3.3                testthat_2.3.2              leaflet_2.0.3              
[184] Matrix_1.2-18               uniftest_1.1                desc_1.2.0                 
[187] RcppAnnoy_0.0.16            TH.data_1.0-10              RColorBrewer_1.1-2         
[190] iterators_1.0.12            ROntoTools_2.14.0           stringr_1.4.0              
[193] htmlwidgets_1.5.1           biomaRt_2.42.0              purrr_0.3.3                
[196] crosstalk_1.1.0.1           globals_0.12.5              openssl_1.4.1              
[199] patchwork_1.0.0             KEGGgraph_1.46.0            codetools_0.2-16           
[202] matrixStats_0.56.0          FNN_1.1.3                   gtools_3.8.1               
[205] EWCE_0.99.2                 prettyunits_1.1.1           SingleCellExperiment_1.8.0 
[208] dbplyr_1.4.2                RSpectra_0.16-0             R.methodsS3_1.8.0          
[211] GenomeInfoDb_1.22.0         gtable_0.3.0                tsne_0.1-3                 
[214] DBI_1.1.0                   stats4_3.6.3                httr_1.4.1                 
[217] highr_0.8                   stringi_1.4.6               progress_1.2.2             
[220] reshape2_1.4.3              english_1.2-5               farver_2.0.3               
[223] viridis_0.5.1               Rgraphviz_2.30.0            ggdendro_0.1-20            
[226] boot_1.3-24                 BiocNeighbors_1.4.1         readr_1.3.1                
[229] bit_1.1-15.2                pkgconfig_2.0.3             gbRd_0.4-11                
[232] knitr_1.28

```

**Comment by brgew** [comment]
Hi,

I installed Bioconductor 3.10 (the current version), and then Monocle3 (using devtools::install_github('cole-trapnell-lab/monocle3', ref="develop")).

I tried reproducing the error using the commands

expression_matrix <- readRDS(url("http://staff.washington.edu/hpliner/data/cao_l2_expression.rds"))
cell_metadata <- readRDS(url("http://staff.washington.edu/hpliner/data/cao_l2_colData.rds"))
gene_annotation <- readRDS(url("http://staff.washington.edu/hpliner/data/cao_l2_rowData.rds"))
cds <- new_cell_data_set(expression_matrix, cell_metadata = cell_metadata, gene_metadata = gene_annotation)

Monocle3 made the cds without reporting an error.

Are you seeing the error still?

**Comment by nfancy** [comment]
Hi, 
yes, I am still having the same error.

**Comment by hpliner** [comment]
Hi @nfancy,
Can you try reinstalling SingleCellExperiment using 
```
BiocManager::install("SingleCellExperiment")
```
And then reinstall monocle3 using 

```
devtools::install_github('cole-trapnell-lab/monocle3', ref="develop"))
```
?
We've seen this error sometimes when the SingleCellExperiment version is updated but Monocle3 isn't.

**Comment by whtns** [comment]
I'm now unable to resolve this issue by reinstalling as you outline

**Comment by hpliner** [comment]
Can you check that your Bioconductor installs are all valid? Check by running: BiocManager::valid()

**Comment by whtns** [comment]
The develop branch that you suggested resolved this issue for me. The root of the error in my situation was due to an erroneous use of the @reducedDims slot of SingleCellExperiment in a home-made function to convert an integrated seurat object to a celldataset. I think this issue could be resolved in the future by supporting a function to manage this conversion as in https://github.com/cole-trapnell-lab/monocle3/issues/262

**Comment by hpliner** [comment]
Sounds like this is resolved along with #262 Reopen if this is still an issue
